datafusion/functions-nested/src/resize.rs

```diff
@@ -134,6 +136,23 @@ pub(crate) fn array_resize_inner(arg: &[ArrayRef]) -> Result<ArrayRef> {
         return exec_err!("array_resize needs two or three arguments");
     }
 
+    let array = &arg[0];
+
+    // Checks if entire array is null
+    if array.null_count() == array.len() {
```

Comment:
`array` is an array of arrays, right?
what happens if some values are null and some other are not?

For example a query processes some null and non-null arrays
```
> SELECT a FROM (VALUES (make_array(1)), (arrow_cast(NULL, 'List(Int32)')), (make_array(3))) t(a);
+-----+
| a   |
+-----+
| [1] |
|     |
| [3] |
+-----+
```

currently, array_resize converts second array (being null) into an empty array.

```
> SELECT array_resize(a, 1) FROM (VALUES (make_array(1)), (arrow_cast(NULL, 'List(Int32)')), (make_array(3))) t(a);
+----------------------------+
| array_resize(t.a,Int64(1)) |
+----------------------------+
| [1]                        |
| []                         |
| [3]                        |
+----------------------------+
```

Reply:
Thanks for pointing that out, I overlooked that.